test/unit/math/p5.Vector.js

```diff
@@ -93,66 +94,198 @@ suite('p5.Vector', function() {
+          v.rotate(180);
+          expect(v.x).to.be.closeTo(0, 0.01);
+          expect(v.y).to.be.closeTo(-1, 0.01);
+          expect(v.z).to.be.closeTo(0, 0.01);
+        });
+
+        test('should rotate the vector [1, 0, 0] by -90 degrees to [0, -1, 0]', function() {
+          v = myp5.createVector(1, 0, 0);
+          v.rotate(-90);
+          expect(v.x).to.be.closeTo(0, 0.01);
+          expect(v.y).to.be.closeTo(-1, 0.01);
+          expect(v.z).to.be.closeTo(0, 0.01);
+        });
+      });
     });
 
-    test('should have x, y, z rotated to 0, -1, 0 (DEGREES)', function() {
-      v.rotate(180);
-      assert.closeTo(v.x, 0, 0.001);
-      assert.closeTo(v.y, -1, 0.001);
-      assert.closeTo(v.z, 0, 0.001);
+    suite('p5.Vector.rotate() [CLASS]', function() {
+      setup(function() {
+        myp5.angleMode(RADIANS);
+      });
+
+      test('should not change the original object', function() {
+        v = myp5.createVector(1, 0, 0);
+        p5.Vector.rotate(v, Math.PI / 2);
+        expect(v.x).to.equal(1);
+        expect(v.y).to.equal(0);
+        expect(v.z).to.equal(0);
+      });
+
+      test('should rotate the vector [0, 1, 0] by pi radians to [0, -1, 0]', function() {
+        v = myp5.createVector(0, 1, 0);
+        const v1 = p5.Vector.rotate(v, Math.PI);
+        expect(v1.x).to.be.closeTo(0, 0.01);
+        expect(v1.y).to.be.closeTo(-1, 0.01);
+        expect(v1.z).to.be.closeTo(0, 0.01);
+      });
+
+      test('should rotate the vector [1, 0, 0] by -pi/2 radians to [0, -1, 0]', function() {
+        v = myp5.createVector(1, 0, 0);
+        const v1 = p5.Vector.rotate(v, -Math.PI / 2);
+        expect(v1.x).to.be.closeTo(0, 0.01);
+        expect(v1.y).to.be.closeTo(-1, 0.01);
+        expect(v1.z).to.be.closeTo(0, 0.01);
+      });
     });
   });
 
-  suite('p5.prototype.angleBetween()', function() {
+  suite('angleBetween', function() {
+    let v1, v2;
     setup(function() {
-      myp5.angleMode(DEGREES);
+      v1 = new p5.Vector(1, 0, 0);
+      v2 = new p5.Vector(2, 2, 0);
     });
 
-    test('should not trip on rounding issues in 2D space', function() {
-      var v1 = myp5.createVector(-11, -20);
-      var v2 = myp5.createVector(-5.5, -10);
-      var res = v1.angleBetween(v2);
-      //expect(Math.abs(v1.angleBetween(v2))).to.be.closeTo(0, 0.00001);
-      expect(res).to.be.closeTo(0, 0.00001);
+    suite('p5.Vector.prototype.angleBetween() [INSTANCE]', function() {
+      test('should return a Number', function() {
+        const res = v1.angleBetween(v2);
+        expect(typeof res).to.eql('number');
+      });
 
-      var v3 = myp5.createVector(-11, -20);
-      var v4 = myp5.createVector(5.5, 10);
-      expect(Math.abs(v3.angleBetween(v4))).to.be.closeTo(180, 0.00001);
-      expect(v3.angleBetween(v4)).to.be.closeTo(180, 0.00001);
-    });
+      test('should not trip on rounding issues in 2D space', function() {
+        v1 = new p5.Vector(-11, -20);
```

Comment:
Since this is a bit of an old PR I would suggest for us to just test changing these back to using `myp5.createVector()` and see if it is working. If that works fine, we can revert this locally then merge this PR so as not to hold things up for much longer.

Reply:
Half the file already used `new p5.Vector()`, and half used `myp5.createVector()` (one of the many inconsistencies that inspired this query: https://github.com/processing/p5.js/pull/5044#issuecomment-778848159). Something I planned to 

My preference is `new p5.Vector()` because it seems more pure and less stateful. If `myp5` has some influence, we probably don't want that. If it doesn't, why use it? Right?

But ultimately it's not a strong preference. I've checked "allow edits by maintainers". Make whatever changes you want.

Reply:
I don't have a strong opinion either way either, I just wanted to make sure there wasn't anything that wasn't working when using `createVector`. I think since the docs mostly describe using `createVector`, it's maybe best that we make sure we test that primarily, although it's also good to know that `new p5.Vector` works too!

Thanks for making the other updates, I'll pull your branch and double check that `createVector` works.

Reply:
Just confirmed that `createVector` works locally! I'll merge your PR as-is though, you're right that the tests are currently very inconsistent with what form it uses, but it does seem to use `new p5.Vector` slightly more in the tests, so this adds a bit more consistency (and we can always discuss standardizing on something else in another issue.)